internal/boxcli/pull.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package boxcli
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go.jetpack.io/devbox"
+	"go.jetpack.io/devbox/internal/pullbox/git"
+)
+
+type pullCmdFlags struct {
+	config configFlags
+	force  bool
+}
+
+func pullCmd() *cobra.Command {
+	flags := pullCmdFlags{}
+	cmd := &cobra.Command{
+		Use:     "pull <file> | <url>",
+		Short:   "Pull a config from a file or URL",
+		Long:    "Pull a config from a file or URL. URLs must be prefixed with 'http://' or 'https://'.",
+		PreRunE: ensureNixInstalled,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return pullCmdFunc(cmd, args, flags.force)
+		},
+		Args: cobra.ExactArgs(1),
+	}
+
+	cmd.Flags().BoolVarP(
+		&flags.force, "force", "f", false,
+		"Force overwrite of existing global config files",
+	)
+
+	flags.config.register(cmd)
```

Comment:
I'm confused. Why add this here, but then remove it from the global command? Did  you also intend to add this `pull` command to the regular commands in `root.go` ?

Reply:
ohh is it because you're calling `addCommandAndHideConfigFlag`. Instead, could you directly call `globalCmd.AddCommand(pullCmd())` ?

Reply:
N/A